common/src/main/java/com/wynntils/features/chat/GuildRankReplacementFeature.java

```diff
@@ -0,0 +1,97 @@
+import com.wynntils.handlers.chat.event.ChatMessageReceivedEvent;
+import com.wynntils.utils.type.IterationDecision;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.minecraft.ChatFormatting;
+import net.minecraft.network.chat.Style;
+import net.minecraftforge.eventbus.api.EventPriority;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+@ConfigCategory(Category.CHAT)
+public class GuildRankReplacementFeature extends Feature {
+    @RegisterConfig
+    public final Config<RankType> rankType = new Config<>(RankType.STARS);
+
+    // Test suite: https://regexr.com/7e5gr
+    private static final Pattern RANK_STARS_PATTERN = Pattern.compile("(§3\\[(?:§b)?)(★{0,5})((?:§3)?.{1,16}])");
+
+    @SubscribeEvent(priority = EventPriority.HIGH)
+    public void onChatMessageReceived(ChatMessageReceivedEvent e) {
+        if (rankType.get() == RankType.STARS) return; // default wynn behavior is stars
+
+        StyledText styledText = e.getStyledText();
+
+        Matcher m = styledText.getMatcher(RANK_STARS_PATTERN);
+        if (!m.find()) return;
+
+        if (rankType.get() == RankType.NONE) {
+            StyledText modified = styledText.iterate((part, changes) -> {
+                if (part.getString(null, PartStyle.StyleType.NONE).contains("★")) {
+                    changes.remove(part);
+                }
+                return IterationDecision.CONTINUE;
+            });
+            e.setMessage(modified.getComponent());
+            return;
+        }
+
+        // rankType == RankType.NAME
+        StyledText modified = styledText.iterateBackwards((part, changes) -> {
+            int stars = (int) part.getString(null, PartStyle.StyleType.NONE)
+                    .chars()
+                    .filter(c -> c == '★')
+                    .count();
+            if (stars > 0) { // This is the part to replace
+                changes.remove(part);
+
+                String rankName = "";
+                switch (stars) {
+                    case 1 -> rankName = "Recruiter ";
+                    case 2 -> rankName = "Captain ";
+                    case 3 -> rankName = "Strategist ";
+                    case 4 -> rankName = "Chief ";
+                    case 5 -> rankName = "Owner ";
+                }
+                changes.add(new StyledTextPart(rankName, part.getPartStyle().getStyle(), null, Style.EMPTY));
+                return IterationDecision.BREAK;
+            }
+
+            if (part.getString(null, PartStyle.StyleType.NONE).contains("[")) {
+                // We've already gone through the rest of the string and did not find any stars, this must be a recruit
+                // Since the [ and the username are the same part, we need to do some surgery:
+                // Find username from the original part, then remove it, and add our new reconstructed one
+                String username = part.getString(null, PartStyle.StyleType.NONE).substring(1);
```

Comment:
Writing code like this is waiting to be broken. Your check doesn't guarantee that the substring will give a meaningful username part. 

I would rather you make a regex for the name part (with colors), and make the username a group. That way even if the regex breaks/format changes we won't error out and worst case we don't add "Recruit".